SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -429,105 +429,61 @@ bool FSpatialGDKEditorToolbarModule::GenerateDefaultLaunchConfig(const FString&
 {
 	FString Text;
 	TSharedRef< TJsonWriter<> > Writer = TJsonWriterFactory<>::Create(&Text);
-	bool bUsingQBI = GetDefault<USpatialGDKSettings>()->bUsingQBI;
+
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+
+	if (SpatialGDKSettings == nullptr)
+	{
+		return false;
+	}
+
+	const FSpatialLaunchConfigDescription& LaunchConfigDescription = SpatialGDKSettings->LaunchConfigDesc;
 
 	// Populate json file for launch config
 	Writer->WriteObjectStart(); // Start of json
-		Writer->WriteValue(TEXT("template"), TEXT("small")); // Template section
+		Writer->WriteValue(TEXT("template"), LaunchConfigDescription.Template); // Template section
 		Writer->WriteObjectStart(TEXT("world")); // World section begin
 			Writer->WriteObjectStart(TEXT("dimensions"));
-				Writer->WriteValue(TEXT("x_meters"), 2000);
-				Writer->WriteValue(TEXT("z_meters"), 2000);
+				Writer->WriteValue(TEXT("x_meters"), LaunchConfigDescription.World.Dimensions.X);
+				Writer->WriteValue(TEXT("z_meters"), LaunchConfigDescription.World.Dimensions.Y);
 			Writer->WriteObjectEnd();
-			Writer->WriteValue(TEXT("chunk_edge_length_meters"), 50);
-			Writer->WriteValue(TEXT("streaming_query_interval"), 4);
+			Writer->WriteValue(TEXT("chunk_edge_length_meters"), LaunchConfigDescription.World.ChunkEdgeLenghtMeters);
+			Writer->WriteValue(TEXT("streaming_query_interval"), LaunchConfigDescription.World.StreamingQueryInterval);
 			Writer->WriteArrayStart(TEXT("legacy_flags"));
-				Writer->WriteObjectStart();
-					Writer->WriteValue(TEXT("name"), TEXT("streaming_query_diff"));
-					Writer->WriteValue(TEXT("value"), TEXT("true"));
-				Writer->WriteObjectEnd();
-				Writer->WriteObjectStart();
-					Writer->WriteValue(TEXT("name"), TEXT("bridge_qos_max_timeout"));
-					Writer->WriteValue(TEXT("value"), TEXT("0"));
-				Writer->WriteObjectEnd();
-				Writer->WriteObjectStart();
-					Writer->WriteValue(TEXT("name"), TEXT("bridge_soft_handover_enabled"));
-					Writer->WriteValue(TEXT("value"), TEXT("false"));
-				Writer->WriteObjectEnd();
-				Writer->WriteObjectStart();
-					Writer->WriteValue(TEXT("name"), TEXT("enable_chunk_interest"));
-					Writer->WriteValue(TEXT("value"), bUsingQBI ? TEXT("false") : TEXT("true"));
-				Writer->WriteObjectEnd();
+			for (auto& Flag : LaunchConfigDescription.World.LegacyFlags)
```

Comment:
i'm curious the GDK teams thoughts on the use of the 'auto' keyword.  The Unreal coding standard suggests avoiding it in C++ unless the iterator type would be very verbose and could impair understanding.  This is just more a question from me on how we handle these :)

Reply:
Isn't this an iterator type, will have a look at what the expanded type would be anyway:)